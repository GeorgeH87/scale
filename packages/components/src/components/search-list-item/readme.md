# scale-search-list-item



<!-- Auto Generated Below -->


## Shadow Parts

| Part                | Description |
| ------------------- | ----------- |
| `"base"`            |             |
| `"label"`           |             |
| `"prefix"`          |             |
| `"suffix"`          |             |
| `"supporting-text"` |             |
| `"text"`            |             |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
