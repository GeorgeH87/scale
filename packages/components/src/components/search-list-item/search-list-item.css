:host {
  --background-hover: var(--telekom-color-ui-state-fill-hovered);
  --text-item-color: var(--telekom-color-text-and-icon-standard);
  --text-supporting-text-color: var(--telekom-color-text-and-icon-additional);
  --font-weight: var(--telekom-typography-font-weight-medium);
  --letter-spacing-standard: var(--telekom-typography-letter-spacing-standard);
  --font-size-supporting-text: 12px;
  --font-size-item-text: 12px;
}

[part~='base'] {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  width: 100%;
  padding: 8px 8px 8px 16px;
  font-family: inherit;
  font-weight: var(--font-weight);
  letter-spacing: var(--letter-spacing-standard);
}

[part~='base'].scale-search-list-item-hover:hover {
  background-color: var(--background-hover);
}

[part~='label'],
[part~='suffix'] {
  display: flex;
}

[part~='prefix'] {
  align-self: flex-start;
}

[part~='label'] {
  font-size: var(--font-size-item-text);
  color: var(--text-item-color);
  padding-top: 8px;
  padding-bottom: 8px;
  text-align: left;
  line-height: 100%;
}

[part~='supporting-text'] {
  font-size: var(--font-size-supporting-text);
  color: var(--text-supporting-text-color);
  line-height: 16px;
  text-align: left;
}

[part~='suffix']:last-of-type {
  margin-left: auto;
}

::slotted([slot='prefix']) {
  display: inline;
  padding-right: 8px;
  margin-top: 4px;
}

::slotted([slot='supporting-text']) {
  display: inline;
}
