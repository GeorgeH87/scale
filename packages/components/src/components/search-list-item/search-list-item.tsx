import { Component, Element, h } from '@stencil/core';

@Component({
  tag: 'scale-search-list-item',
  styleUrl: 'search-list-item.css',
  shadow: true,
})
export class SearchSelectItem {
  @Element() hostElement: HTMLElement;

  render() {
    return (
      <div part="base">
        <div part="prefix">
          <slot name="prefix"></slot>
        </div>

        <div part="text">
          <div part="label">
            <slot></slot>
          </div>
          <div part="supporting-text">
            <slot name="supporting-text"></slot>
          </div>
        </div>

        <div part="suffix">
          <slot name="suffix"></slot>
        </div>
      </div>
    );
  }
}
