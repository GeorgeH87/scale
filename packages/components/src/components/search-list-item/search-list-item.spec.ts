import { newSpecPage } from '@stencil/core/testing';
import { SearchSelectItem } from './search-list-item';

describe('SearchSelectItem', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [SearchSelectItem],
      html: `<scale-search-list-item>Item 01</scale-search-list-item>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
