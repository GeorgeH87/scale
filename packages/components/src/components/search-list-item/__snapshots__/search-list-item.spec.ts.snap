// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchSelectItem should match snapshot 1`] = `
<scale-search-list-item>
  <mock:shadow-root>
    <div class="scale-search-list-item-hover" part="base" tabindex="0">
      <div part="prefix">
        <slot name="prefix"></slot>
      </div>
      <div part="text">
        <div part="label">
          <slot></slot>
        </div>
        <div part="supporting-text">
          <slot name="supporting-text"></slot>
        </div>
      </div>
      <div part="suffix">
        <slot name="suffix"></slot>
      </div>
    </div>
  </mock:shadow-root>
  Item 01
</scale-search-list-item>
`;
